bodhi/tests/client/test_bindings.py

```diff
@@ -24,6 +25,435 @@
+        self.assertEqual(client.login.call_count, 0)
+        client.send_request.assert_called_once_with('csrf', verb='GET', auth=True)
+
+    def test_without_csrf_token_without_cookies(self):
+        """
+        Test the method when csrf_token is not set and has_cookies() returns False.
+        """
+        client = bindings.BodhiClient(username='bowlofeggs', password='illnevertell')
+        client.has_cookies = mock.MagicMock(return_value=False)
+        client.login = mock.MagicMock(return_value='login successful')
+        client.send_request = mock.MagicMock(return_value={'csrf_token': 'a great token'})
+
+        csrf = client.csrf()
+
+        self.assertEqual(csrf, 'a great token')
+        self.assertEqual(client.csrf_token, 'a great token')
+        client.has_cookies.assert_called_once_with()
+        client.login.assert_called_once_with('bowlofeggs', 'illnevertell')
+        client.send_request.assert_called_once_with('csrf', verb='GET', auth=True)
+
+
+class TestBodhiClient_latest_builds(unittest.TestCase):
+    """
+    Test the BodhiClient.latest_builds() method.
+    """
+    def test_latest_builds(self):
+        """
+        Test latest_builds().
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='bodhi-2.4.0-1.fc25')
+
+        latest_builds = client.latest_builds('bodhi')
+
+        self.assertEqual(latest_builds, 'bodhi-2.4.0-1.fc25')
+        client.send_request.assert_called_once_with('latest_builds', params={'package': 'bodhi'})
+
+
+class TestBodhiClient_list_overrides(unittest.TestCase):
+    """
+    Test the BodhiClient.list_overrides() method.
+    """
+    def test_with_user(self):
+        """
+        Test with the user parameter.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='response')
+
+        response = client.list_overrides(user='bowlofeggs')
+
+        self.assertEqual(response, 'response')
+        client.send_request.assert_called_once_with('overrides/', verb='GET',
+                                                    params={'user': 'bowlofeggs'})
+
+    def test_without_user(self):
+        """
+        Test without the user parameter.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='response')
+
+        response = client.list_overrides()
+
+        self.assertEqual(response, 'response')
+        client.send_request.assert_called_once_with('overrides/', verb='GET', params={})
+
+
+class TestBodhiClient_override_str(unittest.TestCase):
+    """
+    Test the BodhiClient.override_str() method.
+    """
+    def test_with_dict(self):
+        """
+        Test override_str() with a dict argument.
+        """
+        override = bindings.BodhiClient.override_str('this is an override')
+
+        self.assertEqual(override, 'this is an override')
+
+    def test_with_str(self):
```

Comment:
I think you got these mixed up, this one uses a dict, the other one uses a string